tokio/src/coop.rs

```diff
@@ -147,9 +128,45 @@ cfg_rt! {
     }
 }
 
-cfg_coop! {
-    use std::task::{Context, Poll};
+pin_project! {
+    /// Future for the [`unconstrained`](unconstrained) method.
+    #[must_use = "Unconstrained does nothing unless polled"]
+    pub struct Unconstrained<F> {
+        #[pin]
+        inner: F,
+    }
+}
+
+impl<F> Future for Unconstrained<F>
+where
+    F: Future,
+{
+    type Output = <F as Future>::Output;
 
+    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        let inner = self.project().inner;
+        with_budget(Budget::unconstrained(), || inner.poll(cx))
+    }
+}
+
+/// Turn off cooperative scheduling for a future. The future or stream will never yield.
```

Comment:
I think we want to give some more detail here, especially around the risks of doing so (starvation). I also think "will never yield" isn't quite accurate — I'd phrase it more as "will never be forced to yield".

Reply:
Done!